api/http-api.md

```diff
@@ -0,0 +1,173 @@
+    * Intentionally not taking the run ID in the URL, but should it be `?runId=` query param?
+  * `SignalWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/signal[/{signal_name}]`
+    * Note signal name on the URL is optional but must be escaped. Must be in body otherwise.
+  * `SignalWithStartWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/signal-with-start`
+  * `ResetWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/reset`
+  * `TerminateWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/terminate`
+  * `ListOpenWorkflowExecutions` - Intentionally not exposed
+  * `ListClosedWorkflowExecutions` - Intentionally not exposed
+  * `ListWorkflowExecutions` - `GET /api/v1/namespaces/{namespace}/workflows`
+    * Accepts `?query=` query parameter
+  * `ListArchivedWorkflowExecutions` - `GET /api/v1/namespaces/{namespace}/archived-workflows`
+    * Intentionally did not do `/workflows/archived` due to ambiguity
+  * `CountWorkflowExecutions` - `GET /api/v1/namespaces/{namespace}/workflow-count`
+  * `QueryWorkflow` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/query[/{query_name}]`
+    * Note query name on the URL is optional but must be escaped. Must be in body otherwise.
+  * `DescribeWorkflowExecution` - `GET /api/v1/namespaces/{namespace}/workflows/{workflow_id}`
+  * `UpdateWorkflowExecution`- `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/updates/{update_id}`
+    * Decided to make this an identity so we can have "describe update" or "cancel update" or whatever one day
+  * `PollWorkflowExecutionUpdate` - Intentionally not exposed
+    * Not sure if best to have long HTTP API here. Would prefer non-blocking describe and maybe one day the difference
+      between describing an update and polling for an update is a "wait" boolean.
+* Activities:
+  * `RecordActivityTaskHeartbeat` - `POST /api/v1/namespaces/{namespace}/activities/heartbeat`
+    * We consider "heartbeat" a verb for these purposes
+  * `RecordActivityTaskHeartbeatById` - `POST /api/v1/namespaces/{namespace}/activities/heartbeat-by-id`
+    * We are intentionally not doing `/activities/{activity_id}/heartbeat` or similar due to URL ambiguity and that we
+      want to encourage the task token approach
+  * `RespondActivityTaskCompleted` - `POST /api/v1/namespaces/{namespace}/activities/complete`
+  * `RespondActivityTaskCompletedById` - `POST /api/v1/namespaces/{namespace}/activities/complete-by-id`
+  * `RespondActivityTaskFailed` - `POST /api/v1/namespaces/{namespace}/activities/fail`
+  * `RespondActivityTaskFailedById` - `POST /api/v1/namespaces/{namespace}/activities/fail-by-id`
+  * `RespondActivityTaskCanceled` - `POST /api/v1/namespaces/{namespace}/activities/cancel`
+  * `RespondActivityTaskCanceledById` - `POST /api/v1/namespaces/{namespace}/activities/cancel-by-id`
+* Search attributes:
```

Comment:
We won't be able to use this API if this is not exposed.

Reply:
:+1: Which calls specifically are being used here? I wonder if I should expose `operatorservice` ones that are better for this I believe.